orttraining/orttraining/core/session/training_session.cc

```diff
@@ -1679,7 +1679,7 @@ Status PipelineTrainingSession::BuildLossAndLossScaling(
     std::string& loss_name,
     optional<std::string>& loss_scale_input_name,
     optional<TrainingConfigurationResult::MixedPrecisionConfigurationResult>& mixed_precision_config_result) {
-  const bool last_pipeline_stage = pipeline_stage_id + 1 == distributed_config.value().pipeline_parallel_size;
+  const bool last_pipeline_stage = pipeline_stage_id == -1 || (pipeline_stage_id + 1 == distributed_config.value().pipeline_parallel_size);
```

Comment:
When `pipeline_stage_id` is `-1`, it means PP is not enabled so every processes have the `last stage`, which is the entire model

Reply:
Seems -1 is a magic number, better to replace with an enum.